compiler/src/iree/compiler/InputConversion/Common/QuantizedMatmulToMatmul.cpp

```diff
@@ -28,51 +28,77 @@ bool isConstantZero(Value val) {
   return constIntOp && constIntOp.value() == 0;
 }
 
-// Pattern lowering quantized_matmul to matmul.
-// Always succeeds.
-//
+// Pattern lowering quantized_matmul to matmul and quantized_batch_matmul to
+// batch_matmul op.
 // This is implementing the math explained in Section 2.3 of
 // https://arxiv.org/abs/1712.05877.
 struct QuantizedMatmulToMatmul
-    : public OpRewritePattern<linalg::QuantizedMatmulOp> {
-  using OpRewritePattern<linalg::QuantizedMatmulOp>::OpRewritePattern;
+    : public OpInterfaceRewritePattern<linalg::LinalgOp> {
+  using OpInterfaceRewritePattern<linalg::LinalgOp>::OpInterfaceRewritePattern;
 
-  LogicalResult matchAndRewrite(linalg::QuantizedMatmulOp quantizedMatmulOp,
+  LogicalResult matchAndRewrite(linalg::LinalgOp op,
                                 PatternRewriter &rewriter) const override {
-    Location loc = quantizedMatmulOp.getLoc();
+    Location loc = op.getLoc();
+    ValueRange inputs{}, outputs{};
+    bool hasBatchDim = false;
+    if (auto qMatmulOp = dyn_cast<linalg::QuantizedMatmulOp>(&op)) {
+      inputs = qMatmulOp->getInputs();
+      outputs = qMatmulOp->getOutputs();
+    } else if (auto qBMatmulOp =
+                   dyn_cast<linalg::QuantizedBatchMatmulOp>(&op)) {
+      inputs = qBMatmulOp->getInputs();
+      outputs = qBMatmulOp->getOutputs();
+      hasBatchDim = true;
+      // Fails when the operation is neither quantized_matmul or
+      // quantized_batch_matmul.
+    } else
+      return failure();
```

Comment:
Please add `{...}` braces also for the `else`.

(And with this line, I realize that you do have a strong need for a if-else chain here as you need a place to return this failure if it's not one of the expected op types. +1 keep that.)

Reply:
We can get away with the chain with the help of `op.getDpsInputs` and `op.getDpsInits`. I have made it a bit simpler.

Reply:
style nit: We wrap single if-statements with braces in IREE. That's the few difference coding style between IREE and MLIR.

Reply:
I see. Thanks, I added that in.